src/coreEnforcer.ts

```diff
@@ -479,7 +479,11 @@ export class CoreEnforcer {
           }
         }
 
-        const [res, done] = effectStream.pushEffect(eftRes);
+        const [res, rec, done] = effectStream.pushEffect(eftRes);
+
+        if (rec) {
+          explainIndex = i;
+        }
 
         if (done) {
           explainIndex = i;
```

Comment:
```suggestion

```

Reply:
When you remove this line, this PR will be merged into master branch.

Reply:
sorry @nodece I have removed it